### PR TITLE
feat(api-server): add task querying, sorting and pagination

### PR DESCRIPTION
**Pull request by koonpeng**
## What's new

The get tasks endpoint now supports querying, results are returned by pages. This allows the webapp to scale properly when there are a large amount of tasks, previously it would have to download every task at once. It is also helpful when we add more specialised view, e.g. a robot view that shows the robot's current task (old api would need to download every task and do client side filtering).

image

## Self-checks

- [ ] I'm familiar with and follow this [Typescript guideline](https://basarat.gitbook.io/typescript/styleguide)
- [ ] I added unit-tests for new components
- [ ] I tried testing edge cases
- [ ] I tested the behavior of the components that interact with the backend, with an e2e test

### CODE REVIEW

File: packages/api-server/api_server/routes/test_base_route.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
+import asyncio
+import concurrent.futures
+
+from ..models import TaskSummary
+from ..models import tortoise_models as ttm
+from .test_fixtures import RouteFixture
+
+
+class TestBaseQuery(RouteFixture):
```
Comment (matiasbavera) [inline packages/api-server/api_server/routes/test_base_route.py:9]: You are testing the Pagination as well here, can we add a comment or add that reference to the test name?

File: packages/dashboard/src/components/tasks/task-page.tsx [hunk 2]
```diff
@@ -14,59 +13,58 @@ const useStyles = makeStyles((theme) => ({
...
+        undefined,
+        undefined,
+        undefined,
+        undefined,
+        undefined,
+        undefined,
+        limit,
+        offset,
+        'priority,-start_time',
+      );
```
Comment (matiasbavera) [inline packages/dashboard/src/components/tasks/task-page.tsx:42]: Just to check is it possible to change `getTasksTasksGet` to an object? Or wrap it? 

e.g: await tasksApi.getTasksTasksGet({limit, offset, options: 'priority,-start_time'}

**Review by matiasbavera — APPROVED** [review]
LGTM, left a couple of comments. 

Once this is merged, maybe @vallq will have a few conflicts to solve on her PR.

File: packages/dashboard/src/components/tasks/task-page.tsx (hunk 2 above)
Comment (koonpeng) [reply packages/dashboard/src/components/tasks/task-page.tsx:42]: unfortunately that is generated from swagger codegen, i havent found a way to change the args yet.

File: packages/api-server/api_server/routes/test_base_route.py (hunk 1 above)
Comment (koonpeng) [reply packages/api-server/api_server/routes/test_base_route.py:9]: I was intending the "base query" to include pagination and sorting together. It was meant to test the "base query" dependency.

**Pull request merged**